tests/nrf5340_audio/lc3_streamer/src/main.c

```diff
@@ -0,0 +1,493 @@
+	zassert_equal(-ENODEV, ret,
+		      "lc3_streamer_register_stream should return -ENODEV from lc3_file");
+	zassert_equal(-1, streamer_idx, "lc3_streamer_register_stream should return index -1");
+	zassert_equal(1, lc3_file_frame_get_fake.call_count,
+		      "lc3_file_frame_get should be called once");
+	zassert_equal(1, lc3_file_open_fake.call_count, "lc3_file_open should be called once");
+	zassert_equal(1, lc3_file_close_fake.call_count, "lc3_file_close should be called once");
+
+	int num_active = lc3_streamer_num_active_streams();
+
+	zassert_equal(0, num_active, "Number of active streams should be 0");
+}
+
+ZTEST(lc3_streamer, test_lc3_streamer_register_stream_invalid_no_frames)
+{
+	int ret;
+	ssize_t streamer_idx;
+
+	lc3_file_frame_get_fake.return_val = -ENODATA;
+
+	ret = lc3_streamer_register_stream("test", &streamer_idx, false);
+	zassert_equal(-ENODATA, ret, "lc3_streamer_register_stream should return -ENODATA");
+	zassert_equal(-1, streamer_idx, "lc3_streamer_register_stream should return index -1");
+	zassert_equal(1, lc3_file_frame_get_fake.call_count,
+		      "lc3_file_frame_get should be called once");
+	zassert_equal(1, lc3_file_open_fake.call_count, "lc3_file_open should be called once");
+	zassert_equal(1, lc3_file_close_fake.call_count, "lc3_file_close should be called once");
+
+	int num_active = lc3_streamer_num_active_streams();
+
+	zassert_equal(0, num_active, "Number of active streams should be 0");
+}
+
+ZTEST(lc3_streamer, test_lc3_streamer_next_frame_get_valid)
+{
+	int ret;
+	ssize_t streamer_idx;
+	uint8_t *frame_buffer_1 = NULL;
+	uint8_t *frame_buffer_2 = NULL;
+	uint8_t *frame_buffer_3 = NULL;
+
+	lc3_file_frame_get_fake.custom_fake = lc3_file_frame_get_fake_valid;
+	k_work_submit_to_queue_fake.custom_fake = k_work_submit_to_queue_valid_fake;
+	k_work_init_fake.custom_fake = k_work_init_valid_fake;
+
+	ret = lc3_streamer_register_stream("test", &streamer_idx, false);
+	zassert_equal(0, ret, "lc3_streamer_register_stream should return success");
+	zassert_equal(0, streamer_idx, "lc3_streamer_register_stream should return index 0");
+
+	ret = lc3_streamer_next_frame_get(streamer_idx, &frame_buffer_1);
+	zassert_equal(0, ret, "lc3_streamer_next_frame_get should return success");
+	zassert_mem_equal(fake_dataset1_valid, frame_buffer_1, fake_dataset1_valid_size,
+			  "Frame data was not as expected");
+
+	ret = lc3_streamer_next_frame_get(streamer_idx, &frame_buffer_2);
+	zassert_equal(0, ret, "lc3_streamer_next_frame_get should return success");
+	zassert_mem_equal(fake_dataset1_valid, frame_buffer_2, fake_dataset1_valid_size,
+			  "Frame data was not as expected");
+
+	ret = lc3_streamer_next_frame_get(streamer_idx, &frame_buffer_3);
+	zassert_equal(0, ret, "lc3_streamer_next_frame_get should return success");
+	zassert_mem_equal(fake_dataset1_valid, frame_buffer_3, fake_dataset1_valid_size,
+			  "Frame data was not as expected");
+
+	zassert_not_equal(frame_buffer_1, frame_buffer_2,
```

Comment:
My thinking was that as the lc3_streamer doesn't really touch the data we don't need to validate here. Looking at it again it might be useful to verify that the pointer returned is correct though.

Let's discuss this together with the other data-validation-discussion!

Reply:
I forgot this myself - but the memory is checked right after each call to `lc3_streamger_next_frame_get()` ( see line 205), so this is already handled :)